### PR TITLE
[SPARK-38469][CORE] Use error class in org.apache.spark.network

### PR DESCRIPTION
**Pull request by bozhang2820**
### What changes were proposed in this pull request?
This PR aims to change exceptions created in package org.apache.spark.netrowk to use error class. This also adds an error class INTERNAL_ERROR_NETWORK and uses that for the internal errors in the package.

### Why are the changes needed?
This is to move exceptions created in package org.apache.spark.network to error class.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Existing tests.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/network/netty/NettyBlockRpcServer.scala [hunk 1]
```diff
@@ -93,8 +94,8 @@ class NettyBlockRpcServer(
           } else {
             val startAndEndId = fetchShuffleBlocks.reduceIds(index)
             if (startAndEndId.length != 2) {
-              throw new IllegalStateException(s"Invalid shuffle fetch request when batch mode " +
-                s"is enabled: $fetchShuffleBlocks")
+              throw SparkException.internalError(s"Invalid shuffle fetch request when batch mode " +
+                s"is enabled: $fetchShuffleBlocks", category = "INTERNAL_ERROR_NETWORK")
```
Comment (MaxGekk) [inline core/src/main/scala/org/apache/spark/network/netty/NettyBlockRpcServer.scala:98]: Just `NETWORK`, I think
```suggestion
                s"is enabled: $fetchShuffleBlocks", category = "NETWORK")
```

File: core/src/main/scala/org/apache/spark/network/netty/NettyBlockRpcServer.scala [hunk 2]
```diff
@@ -93,8 +94,8 @@ class NettyBlockRpcServer(
           } else {
             val startAndEndId = fetchShuffleBlocks.reduceIds(index)
             if (startAndEndId.length != 2) {
-              throw new IllegalStateException(s"Invalid shuffle fetch request when batch mode " +
-                s"is enabled: $fetchShuffleBlocks")
+              throw SparkException.internalError(s"Invalid shuffle fetch request when batch mode " +
```
Comment (MaxGekk) [inline core/src/main/scala/org/apache/spark/network/netty/NettyBlockRpcServer.scala:97]:
```suggestion
              throw SparkException.internalError("Invalid shuffle fetch request when batch mode " +
```

File: core/src/main/scala/org/apache/spark/network/netty/NettyBlockRpcServer.scala [hunk 3]
```diff
@@ -125,8 +126,10 @@ class NettyBlockRpcServer(
         if (blockStored) {
           responseContext.onSuccess(ByteBuffer.allocate(0))
         } else {
-          val exception = new Exception(s"Upload block for $blockId failed. This mostly happens " +
-            s"when there is not sufficient space available to store the block.")
+          val exception = SparkException.internalError(
+            s"Upload block for $blockId failed. This mostly happens " +
+            s"when there is not sufficient space available to store the block.",
```
Comment (MaxGekk) [inline core/src/main/scala/org/apache/spark/network/netty/NettyBlockRpcServer.scala:131]: s is not needed here
```suggestion
            "when there is not sufficient space available to store the block.",
```

File: core/src/main/scala/org/apache/spark/network/netty/NettyBlockRpcServer.scala [hunk 4]
```diff
@@ -125,8 +126,10 @@ class NettyBlockRpcServer(
...
+            category = "INTERNAL_ERROR_NETWORK")
```
Comment (MaxGekk) [inline core/src/main/scala/org/apache/spark/network/netty/NettyBlockRpcServer.scala:132]:
```suggestion
            category = "NETWORK")
```

File: core/src/main/scala/org/apache/spark/network/netty/NettyBlockRpcServer.scala [hunk 5]
```diff
@@ -137,13 +140,15 @@ class NettyBlockRpcServer(
           val errorMsg = "Invalid GetLocalDirsForExecutors request: " +
             s"${if (isIncorrectAppId) s"incorrect application id: ${getLocalDirs.appId};"}" +
             s"${if (execNum != 1) s"incorrect executor number: $execNum (expected 1);"}"
-          responseContext.onFailure(new IllegalStateException(errorMsg))
+          responseContext.onFailure(
+            SparkException.internalError(errorMsg, "INTERNAL_ERROR_NETWORK"))
```
Comment (MaxGekk) [inline core/src/main/scala/org/apache/spark/network/netty/NettyBlockRpcServer.scala:144]:
```suggestion
            SparkException.internalError(errorMsg, category = "NETWORK"))
```

File: core/src/main/scala/org/apache/spark/network/netty/NettyBlockRpcServer.scala [hunk 6]
```diff
@@ -137,13 +140,15 @@ class NettyBlockRpcServer(
...
         } else {
           val expectedExecId = blockManager.asInstanceOf[BlockManager].executorId
           val actualExecId = getLocalDirs.execIds.head
           if (actualExecId != expectedExecId) {
-            responseContext.onFailure(new IllegalStateException(
-              s"Invalid executor id: $actualExecId, expected $expectedExecId."))
+            responseContext.onFailure(SparkException.internalError(
+              s"Invalid executor id: $actualExecId, expected $expectedExecId.",
+              "INTERNAL_ERROR_NETWORK"))
```
Comment (MaxGekk) [inline core/src/main/scala/org/apache/spark/network/netty/NettyBlockRpcServer.scala:151]:
```suggestion
              category = "NETWORK"))
```

**Review by MaxGekk — CHANGES_REQUESTED** [review]

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
Waiting for CI.

**Comment by bozhang2820** [comment]
CI test `single listener, check trigger events are generated correctly` failed, which should be irrelevant to this change?

**Comment by MaxGekk** [comment]
Looking at the two last commit, seems like just flaky tests.

Merging to master. Thank you, @bozhang2820.

**Pull request closed without merging**